roles/kubernetes/master/templates/kubeadm-config.v1alpha1.yaml.j2

```diff
@@ -45,8 +45,12 @@ authorizationModes:
 selfHosted: false
 apiServerExtraArgs:
   bind-address: {{ kube_apiserver_bind_address }}
+{% if kube_apiserver_insecure_port|string != "0" %}
   insecure-bind-address: {{ kube_apiserver_insecure_bind_address }}
+{% endif %}
+{% if kube_apiserver_insecure_port|string != "0" or kube_version | version_compare('v1.10', '<') %}
```

Comment:
Is it acceptable to define both `insecure-bind-address` and `insecure-port` or should this be an else if?

Reply:
@chadswen 
`insecure-port` and `insecure-bind-address` need to be both defined only when `insecure-port`!=0 whatever `kube_version` is.
When `insecure-port`=0, you only need to define it when `kube_version`<1.10.
Maybe I could have set the kube_version condition as a sub-condition of the previous one.

I started with `kube-apiserver.manifest.j2` and didn't want to change the order of the parameters.
I reproduced the same `kubeadm-config.*.yaml.j2`...

I can change that to condition + sub condition and reorder parameters in `kube-apiserver.manifest.j2` if that's what you want, just let me know.

Reply:
Sounds good, please go ahead and do that so we can avoid setting a potentially unsupported param on older kubernetes versions.

Reply:
@chadswen Actually I was wrong in my last comment and adding an `elif` would end up in duplicating lines.
The templates in my PR should work with any version (until the flags are completely removed from apiserver I guess):
- If `insecure-port` is set to 0, there is no need to define `insecure-bind-address` as it won't be used (and will default to 127.0.0.1). Moreover, if `kube_version` is >=1.10, `insecure-port` is deprecated (thus disabled by default) so we don't even need to define `insecure-port`. Setting `insecure-bind-address` with insecure-port` to 0 would be acceptable except for kube-bench that raises a failure (...).
With `kube_version`<1.10, `insecure-port` must be set in order to disable the insecure port.
- If `insecure-port` is not set to 0, both `insecure-bind-address` and `insecure-port` need to be defined whatever `kube_version` is.